pkg/linode-bs/limits.go

```diff
@@ -0,0 +1,31 @@
+package linodebs
+
+// maxVolumeAttachments returns the maximum number of block storage volumes
+// that can be attached to a Linode instance, given the amount of memory the
+// instance has.
+//
+// TODO: This code should be cleaned up to use the built-in max and min
+// functions once the project is updated to Go 1.21. See
+// https://go.dev/ref/spec#Min_and_max.
+func maxVolumeAttachments(memoryBytes uint) int {
```

Comment:
Is this calculation the same as api/vbin? What's the strategy for keeping those two/three things in sync.

Reply:
The calculation isn't exactly the same, only because vbin deals with the memory in gigabytes, and this is working with bytes. However, the results are the same thanks to the tests in `limits_test.go`.

The API does not perform this check unless the volume attachment is persisted to a boot config.

Reply:
Perhaps a vbin PR with a comment to say that the calculation is being relied upon? Between that and e2e tests you'd imagine changes would be caught quite quickly?